metaphor/quick_sight/cll.py

```diff
@@ -0,0 +1,313 @@
+        elif transformation.ProjectOperation:
+            for key in list(columns.keys()):
+                if key not in transformation.ProjectOperation.ProjectedColumns:
+                    columns.pop(key)
+
+        elif transformation.RenameColumnOperation:
+            before = transformation.RenameColumnOperation.ColumnName
+            after = transformation.RenameColumnOperation.NewColumnName
+            columns[after] = columns[before]
+            columns.pop(before)
+
+
+def _process_logical_table_map(
+    resources: Dict[str, ResourceType],
+    tables: Dict[str, Dict[str, Column]],
+    source_entities: Set[str],
+    logical_table_map: Dict[str, DataSetLogicalTable],
+) -> Dict[str, Column]:
+    logical_tables = list(logical_table_map.items())
+    columns: Dict[str, Column]
+
+    while logical_tables:
+        unresolved = []
+        columns = {}
+
+        for table_id, logical_table in logical_tables:
+            source = logical_table.Source
+            if source.DataSetArn:
+                arn = source.DataSetArn
+                upstream_data_set = resources.get(arn)
+                if upstream_data_set is None:
+                    continue
+                assert isinstance(upstream_data_set, DataSet)
+
+                upstream_id = str(
+                    to_entity_id_from_virtual_view_logical_id(
+                        VirtualViewLogicalID(name=arn, type=VirtualViewType.QUICK_SIGHT)
+                    )
+                )
+                source_entities.add(upstream_id)
+                for column in upstream_data_set.OutputColumns or []:
+                    if column.Name is None:
+                        continue
+                    columns[column.Name] = Column(
+                        upstream=[
+                            ColumnReference(upstream_id=upstream_id, name=column.Name)
+                        ]
+                    )
+
+            elif source.PhysicalTableId:
+                upstream_table = tables.get(source.PhysicalTableId)
+                if upstream_table:
+                    columns.update(**upstream_table)
+                else:
+                    assert False, "should not happen"
+
+            elif source.JoinInstruction:
+                left_table_id = source.JoinInstruction.LeftOperand
+                right_table_id = source.JoinInstruction.RightOperand
+
+                left_table = tables.get(left_table_id)
+                right_table = tables.get(right_table_id)
+                if left_table and right_table:
+                    columns.update(**left_table)
+                    columns.update(**right_table)
+
+            if not columns:
+                unresolved.append((table_id, logical_table))
+                continue
+
+            _process_transformation(columns, logical_table.DataTransforms or [])
+
+            tables[table_id] = columns
+
+        logical_tables = unresolved
+    return columns
+
+
+def process_dataset_column_lineage(
```

Comment:
```suggestion
def process_dataset_lineage(
```

It is also doing TLL

Reply:
And rename the file from cll.py to lineage.py ?